lib/check_board.rb

```diff
@@ -3,25 +3,110 @@
 class CheckBoard
   def initialize
     @score = ''
+    @empty_board = [
+      ['-', '-', '-'],
+      ['-', '-', '-'],
+      ['-', '-', '-']
+    ]
   end
 
-  def execute(*board)
-    @check_score = board.reduce
-    if @check_score.nil?
-      @score = ''
+  def execute(board)
+    @check_score = board
+    return @score = '' if @check_score == @empty_board
+    print_score(board)
+  end
+
+  private
+
+  def win_X?
+    diagonal?(:X) || vertical?(:X) || horizontal?(:X)
+  end
+
+  def win_O?
+    diagonal?(:O) || vertical?(:O) || horizontal?(:O)
+  end
+
+  def diagonal?(type)
+    diagonal_forward?(type) || diagonal_backward?(type)
+  end
+
+  def vertical?(type)
+    vertical_left?(type) || vertical_middle?(type) || vertical_right?(type)
+  end
+
+  def horizontal?(type)
+    horizontal_top?(type) || horizontal_middle?(type) || horizontal_bottom?(type)
+  end
+
+  def diagonal_forward?(type)
+    @check_score[0][0] == type && @check_score[1][1] == type && @check_score[2][2] == type
+  end
+
+  def diagonal_backward?(type)
+    @check_score[0][2] == type && @check_score[1][1] == type && @check_score[2][0] == type
+  end
+
+  def vertical_left?(type)
+    @check_score[0][0] == type && @check_score[1][0] == type && @check_score[2][0] == type
+  end
+
+  def vertical_middle?(type)
+    @check_score[0][1] == type && @check_score[1][1] == type && @check_score[2][1] == type
+  end
+
+  def vertical_right?(type)
+    @check_score[0][2] == type && @check_score[1][2] == type && @check_score[2][2] == type
+  end
+
+  def horizontal_top?(type)
+    @check_score[0][0] == type && @check_score[0][1] == type && @check_score[0][2] == type
+  end
 
-    else 
+  def horizontal_middle?(type)
+    @check_score[1][0] == type && @check_score[1][1] == type && @check_score[1][2] == type
+  end
+
+  def horizontal_bottom?(type)
+    @check_score[2][0] == type && @check_score[2][1] == type && @check_score[2][2] == type
+  end
+
+  def print_score_X
+    { status: :X_wins }
+  end
 
-    @check_score.each_with_index do |sub_array , x|
-        sub_array.each_with_index do |item , y|
-         if @check_score[x][y] == :X && @check_score[x+1][y+1] == :X && @check_score[x+2][y+2] == :X
-          return @score = 'X has won'
-         elsif @check_score[x][y] == :O && @check_score[x+1][y+1] == :O && @check_score[x+2][y+2] == :O
-          return @score = 'O has won'
-        end
+  def print_score_O
+    { status: :O_wins }
+  end
+
+  def print_game_over
+    { status: :draw }
+  end
```

Comment:
Do those methods do what their name suggests? Do they print? Or do they only return data?

Reply:
oh! forgot to change after refactoring. Thanks for pointing it out